packages/backend/src/utils/npm.ts

```diff
@@ -194,6 +199,40 @@ class Command {
     return this.execCommand(command);
   }
 
+  public async getNodeProcessVersions(): Promise<NodeJS.ProcessVersions> {
+    try {
+      const output = await this.spawnCommand("node", ["-p", "JSON.stringify(process.versions)"]);
+      return JSON.parse(output);
+    } catch (e) {
+      getConsoleWarnLogger().error(`Error retrieving NodeJS process versions: ${e.message}`);
+      return {} as NodeJS.ProcessVersions;
+    }
+  }
+
+  private spawnCommand(command: string, commandArgs: string[]): Promise<string> {
+    return new Promise((resolve, reject) => {
+      const spawnOptions = /^win/.test(process.platform)
+        ? { windowsVerbatimArguments: true, shell: true, cwd: os.homedir() }
+        : { cwd: os.homedir() };
+      let output = "";
+      const spawnProcess = spawn(command, commandArgs, spawnOptions);
+      spawnProcess.stdout.on("data", (data) => {
+        const newData = data.toString();
+        output += newData;
+      });
+      spawnProcess.stderr.on("data", (data) => {
+        const newData = data.toString();
+        output += newData;
+      });
+      spawnProcess.on("exit", () => {
+        resolve(output);
+      });
+      spawnProcess.on("error", (error) => {
+        reject(error);
```

Comment:
you are fulfilling a promise with a string, but the receiver is expecting an `Error` type. Have you checked this case? Please add an appropriate unit test.

Reply:
Not sure I understand the ask here. The reject will return the error object, reject has type `any` afaik so not sure if there is an expectation of what is returned here. I can return a string/error.message/etc if desired?